can/bus.py

```diff
@@ -2,29 +2,43 @@
 # coding: utf-8
 
 """
-Contains the ABC bus implementation.
+Contains the ABC bus implementation and documentation.
 """
 
 from __future__ import print_function, absolute_import
 
 from abc import ABCMeta, abstractmethod
 import logging
 import threading
+from time import time
 
 from can.broadcastmanager import ThreadBasedCyclicSendTask
 
 logger = logging.getLogger(__name__)
 
 
 class BusABC(object):
-    """CAN Bus Abstract Base Class
 
-    Concrete implementations must implement the following methods:
-        * send
-        * recv
+    """CAN Bus Abstract Base Class.
+
+    Concrete implementations *must* implement the following:
+        * :meth:`~can.BusABC.send`
+        * :meth:`~can.BusABC._recv_internal`
+        * set the :attr:`~can.BusABC.channel_info` attribute to a string describing
+          the interface and/or channel
+
+    The *may* implement the following:
+        * :meth:`~can.BusABC.flush_tx_buffer` to allow discrading any
+          messages yet to be sent
+        * :meth:`~can.BusABC.shutdown` to override how the bus should
+          shut down, in which case the class has to either call through
+          `super().shutdown()` or call :meth:`~can.BusABC.flush_tx_buffer`
+          on its own
```

Comment:
Flushing buffer is usually unnecessary.

Reply:
But it is implemented like that right now in the BusABC class. Shell it be removed from there as well?